Java/core/src/main/java/com/amazon/randomcutforest/returntypes/SampleSummary.java

```diff
@@ -96,33 +101,56 @@ public SampleSummary(double[] point, float weight) {
         this.weightOfSamples = weight;
         this.summaryPoints[0] = toFloatArray(point);
         this.relativeWeight[0] = weight;
+        this.measure[0] = new float[point.length];
         System.arraycopy(this.summaryPoints[0], 0, this.median, 0, point.length);
         System.arraycopy(this.summaryPoints[0], 0, this.mean, 0, point.length);
         System.arraycopy(this.summaryPoints[0], 0, this.upper, 0, point.length);
         System.arraycopy(this.summaryPoints[0], 0, this.lower, 0, point.length);
     }
 
-    void addTypical(float[][] summaryPoints, float[] relativeWeight) {
+    void addTypical(float[][] summaryPoints, float[] relativeWeight, float[][] measure) {
         checkArgument(summaryPoints.length == relativeWeight.length, "incorrect lengths of fields");
+        checkArgument(summaryPoints.length == measure.length, "incorrect lengths of fields");
         if (summaryPoints.length > 0) {
             int dimension = summaryPoints[0].length;
             this.summaryPoints = new float[summaryPoints.length][];
+            this.measure = new float[summaryPoints.length][];
             for (int i = 0; i < summaryPoints.length; i++) {
                 checkArgument(dimension == summaryPoints[i].length, " incorrect length points");
+                checkArgument(dimension == measure[i].length, " incorrect length points");
                 this.summaryPoints[i] = Arrays.copyOf(summaryPoints[i], dimension);
+                this.measure[i] = Arrays.copyOf(measure[i], dimension);
             }
             this.relativeWeight = Arrays.copyOf(relativeWeight, relativeWeight.length);
         }
     }
 
+    public SampleSummary(List<Weighted<float[]>> points, SampleSummary clusters) {
+        this(points, DEFAULT_PERCENTILE);
+        this.addTypical(clusters.summaryPoints, clusters.relativeWeight, clusters.measure);
+    }
+
     public SampleSummary(List<Weighted<float[]>> points, float[][] summaryPoints, float[] relativeWeight,
-            double percentile) {
+            float[] measure, double percentile) {
         this(points, percentile);
-        this.addTypical(summaryPoints, relativeWeight);
+        float[][] transformedMeasure = new float[measure.length][];
+        double factor = 0;
+        for (int j = 0; j < this.upper.length; j++) {
+            factor += max(0, this.upper[j] - this.lower[j]);
+        }
+        for (int i = 0; i < measure.length; i++) {
+            transformedMeasure[i] = new float[this.upper.length];
+            for (int j = 0; j < this.upper.length; j++) {
+                transformedMeasure[i][j] = (factor == 0) ? 0
```

Comment:
It's not used yet -- but it can be very useful. This indicates the "shape" of the box around the representative point -- the relative weight captures the mass, and this can capture the extent.

Reply:
Can you explain "the relative weight captures the mass, and this can capture the extent"? What's mass and what's extent? How is it related to transformedMeasure?

I thought for internal node, mass is the number of leave node. For leaf node, mass is the leaf node's duplicity.